go/pkg/bertymessenger/api.go

```diff
@@ -119,49 +119,8 @@ func (svc *service) internalInstanceShareableBertyID(ctx context.Context, req *I
 	return &ret, nil
 }
 
-func (svc *service) ParseDeepLink(ctx context.Context, req *ParseDeepLink_Request) (*ParseDeepLink_Reply, error) {
-	if req == nil || req.Link == "" {
-		return nil, errcode.ErrMissingInput
-	}
-
-	ret := ParseDeepLink_Reply{}
-
-	query, method, err := NormalizeDeepLinkURL(req.Link)
-	if err != nil {
-		return nil, errcode.ErrInvalidInput.Wrap(err)
-	}
-
-	switch method {
-	case "/id":
-		ret.Kind = ParseDeepLink_BertyID
-		ret.BertyID = &BertyID{}
-		key := query.Get("key")
-		if key == "" {
-			return nil, errcode.ErrMessengerInvalidDeepLink
-		}
-		payload, err := b64DecodeBytes(key)
-		if err != nil {
-			return nil, errcode.ErrMessengerInvalidDeepLink.Wrap(err)
-		}
-		err = proto.Unmarshal(payload, ret.BertyID)
-		if err != nil {
-			return nil, errcode.ErrMessengerInvalidDeepLink.Wrap(err)
-		}
-		if len(ret.BertyID.PublicRendezvousSeed) == 0 || len(ret.BertyID.AccountPK) == 0 {
-			return nil, errcode.ErrMessengerInvalidDeepLink
-		}
-		if name := query.Get("name"); name != "" {
-			ret.BertyID.DisplayName = name
-		}
-
-	case "/group":
-		return ParseGroupInviteURLQuery(query)
-
-	default:
-		return nil, errcode.ErrMessengerInvalidDeepLink
-	}
-
-	return &ret, nil
+func (svc *service) ParseDeepLink(_ context.Context, req *ParseDeepLink_Request) (*ParseDeepLink_Reply, error) {
+	return DecodeDeepLink(req)
```

Comment:
```suggestion
	if req == nil {
		return nil, errcode.ErrMissingInput
	}
	return DecodeDeepLink(req.Link)
```

Reply:
That already done there :
https://github.com/Jorropo/berty/blob/ab9bb6c6d52a3eb9041ee1d291615121dfacc96f/go/pkg/bertymessenger/api.go#L183-L185

Reply:
the suggestion is not about double checking `if req == nil`, but about having a helper that takes a link in parameter instead of a struct only containing the link